lib/common.bash

```diff
@@ -28,9 +31,39 @@ KATA_TESTS_CACHEDIR="${KATA_TESTS_CACHEDIR:-${KATA_TESTS_BASEDIR}/cache}"
 KATA_HYPERVISOR="${KATA_HYPERVISOR:-qemu}"
 experimental_qemu="${experimental_qemu:-false}"
 
+# Display a message to stderr, a dump of lots of useful debug
+# information (including a full stacktrace) and exit 1.
 die() {
 	local msg="$*"
-	echo -e "[$(basename $0):${BASH_LINENO[0]}] ERROR: $msg" >&2
+
+	echo >&2 "ERROR: $msg"
+
+	[ -z "${CI:-}" ] && exit 1
+
+	# We are running in the CI so dump as much information about
+	# the environment that generated the failure as possible.
+
+	# This function is called to indicate a fatal error occurred, so
+	# the caller of this function is the site of the detected error.
+	local error_location
+	error_location=$(caller 0)
```

Comment:
Why split it on a separate line?

Reply:
`shellcheck(1)` complains otherwise:

```
SC2155: Declare and assign separately to avoid masking return values.
```

Reply:
Plus https://github.com/kata-containers/tests/pull/4733/files#r858730128 explains the message.